### PR TITLE
OMO-2002 Corsless GETs

### PR DESCRIPTION
**Pull request by notatestuser**
https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Simple_requests
https://m.alphasights.com/killing-cors-preflight-requests-on-a-react-spa-1f9b04aa5730#4bdf

### CODE REVIEW

File: src/domain/Api/index.js [hunk 1]
```diff
@@ -42,6 +55,24 @@ export const fetch = (url, options = {}) => {
   });
 };
 
+function getDefaultFetchOpts(options) {
+  const isCorsSimpleMethod =
+      is.empty(options) ||
+      (is.object(options) && (
+        is.undefined(options.method) ||
+        (options.method &&
+          CORS_SIMPLE_METHODS.includes(options.method.toUpperCase()))));
```
Comment (rulych) [inline src/domain/Api/index.js]: These few lines are like staring into the minds of a superior alien race.

Comment (notatestuser) [inline src/domain/Api/index.js]: suggestions welcome

Comment (notatestuser) [inline src/domain/Api/index.js]: it can be simplified a bit

File: src/domain/Api/index.js [hunk 2]
```diff
@@ -42,6 +55,24 @@ export const fetch = (url, options = {}) => {
...
+  return {
+    headers: Object.assign(
+      { Accept: 'application/json; charset=utf-8' },
+      ! isCorsSimpleMethod ? {
+        'Content-Type': 'application/json',
```
Comment (rulych) [inline src/domain/Api/index.js]: Have you checked if this could be removed completely, just in case we could simplify this a bit?

Comment (notatestuser) [inline src/domain/Api/index.js]: I've checked, we need it for PUT, POST, DELETE

Comment (rulych) [inline src/domain/Api/index.js]: OK, thanks for the info. 😁

File: src/domain/Api/index.js (hunk 1 above)
Comment (rulych) [inline src/domain/Api/index.js]: Maybe we could at least put into a separate function to make it more vertical and readable?

Comment (notatestuser) [inline src/domain/Api/index.js]: see 10100a4

File: src/domain/Api/index.js [hunk 3]
```diff
@@ -42,6 +55,23 @@ export const fetch = (url, options = {}) => {
   });
 };
 
+function getDefaultFetchOpts(options) {
+  const isCorsSimpleMethod =
+      is.empty(options) ||
+      is.falsy(options.method) ||
+      (is.string(options.method) &&
+        CORS_SIMPLE_METHODS.includes(options.method.toUpperCase()));
```
Comment (rulych) [inline src/domain/Api/index.js]: Oh, OK, this is not that bad now. 👍

File: src/domain/Api/index.js (hunk 2 above)
Comment (notatestuser) [inline src/domain/Api/index.js]: no problem。

**Pull request merged**